File: 프로그래머스/1/12925. 문자열을 정수로 바꾸기/문자열을 정수로 바꾸기.kt
class Solution {
    fun solution(s: String): Int {
        var sign = 1
        var start = 0
        var answer = 0

        if (s[0] == '-') {
            sign = -1
            start = 1
        } else if (s[0] == '+') {
            start = 1
        }

        for (i in start until s.length) {
            answer = answer * 10 + (s[i] - '0')
        }

        return sign * answer
    }
}

File: 프로그래머스/1/12925. 문자열을 정수로 바꾸기/문자열을 정수로 바꾸기.js
function solution(s) {
    let sign = 1;
    let start = 0;
    let answer = 0;

    if (s[0] === '-') {
        sign = -1;
        start = 1;
    } else if (s[0] === '+') {
        start = 1;
    }

    for (let i = start; i < s.length; i++) {
        answer = answer * 10 + (s[i] - '0');
    }

    return sign * answer;
}
